api/materialized_views/search/recordNameSubset.js

```diff
@@ -0,0 +1,95 @@
+const mongodb = require('../../src/utils/mongodb');
+
+/**
+ * Updates the recordName subset.
+ *
+ * Note: recordName exists on all 14 basic record types, but is currently only accessible by users in any capacity as
+ * part of the  LNG flavour. Therefore, this subset should not include the recordName field from master or other non-LNG
+ * flavour records.
+ *
+ * @param {*} defaultLog
+ */
+async function update(defaultLog) {
+  const aggregate = [
+    {
+      // Match all master records that could have an LNG flavour record
+      $match: {
+        _schemaName: {
+          $in: [
+            'AdministrativePenalty',
+            'AdministrativeSanction',
+            'Agreement',
+            'Certificate',
+            'ConstructionPlan',
+            'CourtConviction',
+            'Inspection',
+            'ManagementPlan',
+            'Order',
+            'Permit',
+            'RestorativeJustice',
+            'SelfReport',
+            'Ticket',
+            'Warning'
+          ]
+        }
+      }
+    },
+    {
+      // Lookup associated flavour records
+      $lookup: {
+        from: 'nrpti',
+        localField: '_flavourRecords',
+        foreignField: '_id',
+        as: 'flavours'
+      }
+    },
+    {
+      // Unwind array of flavour records
+      $unwind: '$flavours'
+    },
+    {
+      // Redact any records that don't aren't an LNG flavour type
+      $redact: {
+        $cond: {
+          if: {
+            $in: [
+              '$flavours._schemaName',
+              [
+                'AdministrativePenaltyLNG',
+                'AdministrativeSanctionLNG',
+                'AgreementLNG',
+                'CertificateLNG',
+                'ConstructionPlanLNG',
+                'CourtConvictionLNG',
+                'InspectionLNG',
+                'ManagementPlanLNG',
+                'OrderLNG',
+                'PermitLNG',
+                'RestorativeJusticeLNG',
+                'SelfReportLNG',
+                'TicketLNG',
+                'WarningLNG'
+              ]
+            ]
+          },
+          then: '$$KEEP',
+          else: '$$PRUNE'
+        }
+      }
+    }
+  ];
```

Comment:
```
await collection.createIndex(
  {
    'flavours.recordName': 'text'
  },
  {
    name: 'record-name-subset-text-index'
  }
);
```
This is ultimately asking to index against the `recordName` property on LNG records only.  It's very inefficient to query the master records, populate everything, then redact, when you could have matched against LNG records only, insert them into the collection, and then create an index on that field.  Records that don't have recordName wouldn't be returned in the result because the index wouldn't hit, or your API call against this indexed collection could query of $text _and_ _schemaName: 'xxx'.

Reply:
I think we could use this [partial index concept](https://docs.mongodb.com/manual/core/index-partial/) to create text indexes on only LNG schemas.  

And then this subset could be the same as the description_summary one, where we only do a lookup on flavours, and thats it.

Reply:
@marklise I've updated this PR as per Jess's comment (https://bcmines.atlassian.net/browse/NRPT-71).  So it is now just a regular subset on a master field.